tools/xdp/s2n-quic-xdp/src/task/completion_to_tx.rs

```diff
@@ -0,0 +1,456 @@
+            comp,
+            poller,
+            assignment,
+        } = self.get_mut();
+
+        trace!("polling completion ring to tx");
+
+        // try to query if we have any ready items
+        let count = match poller.poll(comp, cx) {
+            Poll::Ready(Some(count)) => {
+                // we're ready, keep going
+                count
+            }
+            Poll::Ready(None) => {
+                // shut down the task
+                return Poll::Ready(());
+            }
+            Poll::Pending => {
+                // nothing to do right now
+                return Poll::Pending;
+            }
+        };
+
+        let (head, tail) = comp.data();
+
+        let mut sent = 0;
+        let mut closed = 0;
+
+        let mut idx = 0;
+        txs.for_each(|tx| {
+            match tx.try_slice() {
+                Ok(Some(mut slice)) => {
+                    /// copies the completion items into the worker's queue
+                    macro_rules! extend {
+                        ($name:ident) => {
+                            if !$name.is_empty() {
+                                let mut iter = $name
+                                    .iter()
+                                    .take(count as _)
+                                    .copied()
+                                    .filter(|desc| assignment.assign(*desc, idx))
+                                    .map(|desc| {
+                                        trace!("assigning address {} to queue {idx}", desc.address);
+
+                                        sent += 1;
+                                        desc
+                                    })
+                                    .peekable();
+
+                                while iter.peek().is_some() {
+                                    if slice.extend(&mut iter).is_err() {
+                                        trace!("tx queue {idx} is closed");
+                                        closed += 1;
+                                        idx += 1;
+                                        return;
+                                    }
+                                }
+                            }
+                        };
+                    }
+
+                    extend!(head);
+                    extend!(tail);
+                }
+                Ok(None) => {
+                    unreachable!("tx queue capacity should exceed that of the completion queue");
+                }
+                Err(_) => {
+                    trace!("tx queue {idx} closed");
+                    closed += 1;
+                }
+            }
+
+            idx += 1;
+        });
+
+        // let the poller know how many items we consumed
+        poller.release(comp, sent);
+
+        // if all of the queues are closed then shut down the task
+        if closed == txs.len() {
+            trace!("all tx queues closed; shutting down");
+            return Poll::Ready(());
+        }
```

Comment:
Initially I had the ordering of these two reversed, which caused the fuzz tests to randomly fail at the end and drop a few descriptors.